Collect TCP statistics

jukkar opened the pull request.

jukkar committed:
net: tcp: Add TCP statistics support

We did not collect any TCP statistics before but this commit
changes that.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

tbursztyka commented on subsys/net/ip/connection.c:
```diff
@@ -871,11 +873,17 @@ enum net_verdict net_conn_input(enum net_ip_protocol proto, struct net_pkt *pkt)
 		net_stats_update_udp_drop();
+	} else if (IS_ENABLED(CONFIG_NET_TCP) && proto == IPPROTO_TCP) {
+		net_stats_update_tcp_seg_drop();
 	}
```
Hum, why not wrapping this if/else if (as well as the update recv above) in an inline function such as: 
net_stats_update_per_proto_recv(proto);
and:
net_stats_update_per_proto_drop(proto);

lpereira replied:
+1 for this

jukkar replied:
Sure, why not. I will cook a new version.

jukkar committed:
net: stats: Add UDP checksum stats update function

A function updating UDP checksum failure statistics was missing.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
New version sent. Added also new patch that adds possibility to update UDP checksum error count.

jukkar committed:
net: stats: Activate RPL and MLD stats if needed

If user has enabled RPL and STATISTICS, then enable RPL
statistics by default as that is probably what user want.
Same thing for MLD statistics.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
net: stats: IPv6 multicast listener daemon stats not printed

The MLD statistics were collected but never printed.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
Added two more statistics related fixes that I noticed while testing the statistics printing.

pfalcon commented on include/net/net_stats.h:
```diff
@@ -90,22 +101,25 @@ struct net_stats_tcp {
+	/** Number of received TCP segments with a bad ACK number. */
 	net_stats_t ackerr;
 
+	/** Number of received TCP bad RST (reset) segments. */
```
Minor nitpick: "TCP bad RST" sounds strange, maybe "bad TCP RST"?

jukkar replied:
sure, I will fix that

pfalcon reviewed (approved):
I'd also find enabling stats by default strange. As a further step on https://jira.zephyrproject.org/browse/ZEP-2105, I wanted to propose to enable NET_ERR logging by default (so the case of no netif's was visibly reported to a user). Assuming we can only have one enabled by default - stats or err logging, I'd prefer logging.

jukkar commented:
> I'd also find enabling stats by default strange.

Please note that we enable only statistics for given subsystem like RPL if both CONFIG_NET_STATISTICS and CONFIG_NET_RPL is defined. IMHO this is quite logical now so that user does not have to separately enable RPL statistics in this case.
